src/Accessor/DefaultAccessorStrategy.php

```diff
@@ -20,21 +20,76 @@
 
 namespace JMS\Serializer\Accessor;
 
+use JMS\Serializer\Exception\ExpressionLanguageRequiredException;
+use JMS\Serializer\Exception\LogicException;
+use JMS\Serializer\Expression\ExpressionEvaluatorInterface;
+use JMS\Serializer\Metadata\ExpressionPropertyMetadata;
 use JMS\Serializer\Metadata\PropertyMetadata;
+use JMS\Serializer\Metadata\StaticPropertyMetadata;
+use JMS\Serializer\Metadata\VirtualPropertyMetadata;
 
 /**
  * @author Asmir Mustafic <goetas@gmail.com>
  */
 final class DefaultAccessorStrategy implements AccessorStrategyInterface
 {
+    private $readAccessors = array();
+    private $writeAccessors = array();
+
+    /**
+     * @var ExpressionEvaluatorInterface
+     */
+    private $evaluator;
+
+    public function __construct(ExpressionEvaluatorInterface $evaluator = null)
+    {
+        $this->evaluator = $evaluator;
+    }
 
     public function getValue(object $object, PropertyMetadata $metadata)
     {
-        return $metadata->getValue($object);
+        if ($metadata instanceof StaticPropertyMetadata) {
```

Comment:
Nice. This bunch of IFs calls for a strategy pattern imho, this is a bit too coupled.

Reply:
this has to be fast! very fast! :thinking: 
not sure if having extra classes to load and call will influence on it